js/src/common/utils/anchorScroll.js

```diff
@@ -12,10 +12,18 @@
  * @param {Function} callback The callback to run that will change page content.
  */
 export default function anchorScroll(element, callback) {
+  const $element = $(element);
+
+  // return if element doesn't exist
```

Comment:
These comments are stating what's pretty obvious in the code - in this case, it's not even 100% correct, because the callback is still executed.

Please explain *why* this is necessary, not *what* it does.

Reply:
This reminds me of Clean Code, chapter 4 😄 🚀

Reply:
@datitisev It seems like you also misunderstood me. I would love to see comments explaining *why* this is needed (because of something that happens before this code is called), not what error it prevents in that same method. That's mostly obvious from looking at the code.

What I don't see when looking at this code:
- Why would the `discussionList` ever not exist when the post composer is open?
- When would somebody pass a not-existing element into `anchorScroll`?

Reply:
But, again, let's focus on the other PRs first.

Reply:
> Why would the discussionList ever not exist when the post composer is open?

I can only assume during redraws or something

> When would somebody pass a not-existing element into anchorScroll?

Not intentionally. If you look at the stack trace, it was because of the PostStream. From there, I assume it tried to scroll to a post (through its index/number) that hadn't been loaded yet.

There's really not much I can gather from the stack traces, and this is a pretty simple fix. This would be a temporary solution, until (or if) we find and & solve the underlying issues.